ldpd/lde.c

```diff
@@ -1533,3 +1566,104 @@ lde_address_list_free(struct lde_nbr *ln)
 		free(lde_addr);
 	}
 }
+
+static void
+lde_del_label_chunk(void *val)
+{
+	free(val);
+}
+static int
+lde_get_label_chunk(void)
+{
+	int ret;
+	uint32_t start, end;
+
+	log_debug("Getting label chunk");
+	ret = lm_get_label_chunk(zclient_sync, 0, CHUNK_SIZE, &start, &end);
+	if (ret < 0)
+	{
+		log_warnx("Error getting label chunk!");
+		close(zclient_sync->sock);
+		zclient_sync->sock = -1;
+		return -1;
+	}
+
+	on_get_label_chunk_response(start, end);
+
+	return 0;
+}
+static void
+lde_label_list_init(void)
+{
+	label_chunk_list = list_new();
+	label_chunk_list->del = lde_del_label_chunk;
+
+	/* get first chunk */
+	while (lde_get_label_chunk() != 0) {
+		fprintf(stderr, "Error getting first label chunk!\n");
+		sleep(1);
+	}
+}
```

Comment:
Same here. A safer mode will be to try 10 times and then exit with error instead of loping indefinitely